ambry-commons/src/test/java/com.github.ambry.commons/BlobIdTest.java

```diff
@@ -425,19 +340,12 @@ private void assertBlobIdLongForm(short version) throws IOException {
   private BlobId getRandomBlobId(short version) throws Exception {
     byte[] bytes = new byte[2];
     random.nextBytes(bytes);
-    byte flag = bytes[0];
     random.nextBytes(bytes);
     byte datacenterId = bytes[0];
     short accountId = getRandomShort(random);
     short containerId = getRandomShort(random);
+    BlobIdType type = random.nextBoolean() ? BlobIdType.NATIVE : BlobIdType.CRAFTED;
     PartitionId partitionId = referenceClusterMap.getWritablePartitionIds().get(random.nextInt(3));
-    switch (version) {
-      case BLOB_ID_V1:
-        return new BlobIdV1(flag, datacenterId, accountId, containerId, partitionId);
-      case BLOB_ID_V2:
-        return new BlobId(flag, datacenterId, accountId, containerId, partitionId);
-      default:
-        throw new Exception("Unrecognized blobId version " + version);
-    }
+    return new BlobId(version, type, datacenterId, accountId, containerId, partitionId);
   }
 }
```

Comment:
any tests for the `compareTo()` method to ensure that only uuid is the one being compared ?